src/multimarginloss.cpp

```
@@ -0,0 +1,149 @@
+        auto tmp   = multimarginloss::InvokeParams{};
+        tmp.type   = InvokeType::Run;
+        tmp.iDesc  = &iDesc;
+        tmp.i      = i;
+        tmp.tDesc  = &tDesc;
+        tmp.t      = t;
+        tmp.wDesc  = &wDesc;
+        tmp.w      = w;
+        tmp.oDesc  = &oDesc;
+        tmp.o      = o;
+        tmp.p      = p;
+        tmp.margin = margin;
+        return tmp;
+    }();
+
+    const auto algo = AlgorithmName{"MultiMarginLossUnreducedForward"};
+    const auto solvers =
+        solver::SolverContainer<solver::multimarginloss::MultiMarginLossUnreducedForward>{};
+
+    solvers.ExecutePrimitive(handle, problem, algo, invoke_params);
+
+    return miopenStatusSuccess;
+}
+
+std::size_t GetMultiMarginLossForwardWorkspaceSize(Handle& handle,
+                                                   const TensorDescriptor& iDesc,
+                                                   const TensorDescriptor& tDesc,
+                                                   const TensorDescriptor& wDesc,
+                                                   const TensorDescriptor& oDesc,
+                                                   const long p,
+                                                   const float margin,
+                                                   miopenLossReductionMode_t reduction)
+{
+    if(reduction == MIOPEN_LOSS_REDUCTION_NONE)
+    {
+        return static_cast<size_t>(0);
+    }
+    auto ctx            = ExecutionContext{&handle};
+    const float divisor = (reduction == MIOPEN_LOSS_REDUCTION_MEAN) ? iDesc.GetLengths()[0] : 1;
+    const auto problem =
+        multimarginloss::ForwardProblemDescription{iDesc, tDesc, wDesc, oDesc, p, margin, divisor};
+
+    const auto solvers = solver::SolverContainer<solver::multimarginloss::MultiMarginLossForward>{};
+
+    auto pair_size_vector = solvers.GetWorkspaceSizes(ctx, problem);
+    return pair_size_vector.empty() ? static_cast<size_t>(-1) : pair_size_vector.front().second;
```

I've noticed that `Get*WorkspaceSize` may return `size_t(-1)` value in case of empty solvers.
I guess it's not quite a good solution because it may lead to a crash while allocating 2^64 bytes of GPU memory.

It's better either to return 0 or to throw an exception that no solution found.

It's just a notice and has to be fixed in a separated PR across all the other affected solvers like Sum or t5layernorm.
@seungmanhan

---

It should not return 0 because 0 means "we need 0 workspace size". For example in my operation when reduction mode = None calling this function will return 0. Maybe throw an exception is a better solution and I agree that this need to be discussed more and fixed in a separated PR.